homeassistant/components/ibeacon/coordinator.py

```diff
@@ -428,6 +451,28 @@ def _async_update_rssi_and_transients(self) -> None:
                     ibeacon_advertisement,
                 )
 
+    async def async_config_entry_updated(
+        self, hass: HomeAssistant, config_entry: ConfigEntry
+    ) -> None:
+        """Restore ignored nameless beacons when the allowlist is updated."""
+
+        for uuid in self._entry.options.get(CONF_ALLOW_NAMELESS_UUIDS, []):
+            for address in self._ignored_nameless_by_uuid.pop(uuid, set()):
+                _LOGGER.debug(
+                    "restoring nameless iBeacon %s from address %s", uuid, address
+                )
+
+                service_info = bluetooth.async_last_service_info(
+                    self.hass, address, connectable=False
+                )
+                if not service_info:
+                    continue  # no longer available
```

Comment:
```suggestion
                if not (service_info := bluetooth.async_last_service_info(
                    self.hass, address, connectable=False
                )):
                    continue  # no longer available
```

Please add missing coverage for this line

Reply:
Do you know how to make `async_last_service_info` return an empty result in a test after `inject_bluetooth_service_info` was used to simulate a beacon?

Btw there's a [similar line](https://github.com/home-assistant/core/blob/918ea8f56d14af7565500e4d210e6bb96b6efae5/homeassistant/components/ibeacon/coordinator.py#L402-L403) that's also not covered.

Reply:
Move time forward enough that the service info expires.

Examples in xiaomi_ble and bthome tests

Reply:
Done.